improver_tests/visibility/test_visibility_combine_cloud_base.py

```diff
@@ -0,0 +1,251 @@
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Unit tests for VisibilityCombineCloudBase plugin"""
+
+from datetime import datetime
+
+import iris
+import numpy as np
+import pytest
+from iris.cube import Cube, CubeList
+
+from improver.synthetic_data.set_up_test_cubes import set_up_probability_cube
+from improver.utilities.probability_manipulation import invert_probabilities
+from improver.visibility.visibility_combine_cloud_base import VisibilityCombineCloudBase
+
+cloud_name = (
+    "cloud_base_assuming_only_consider_cloud_area_fraction_greater_than_4p5_oktas"
+)
+
+
+@pytest.fixture()
+def visibility_gridded_cube() -> Cube:
+    """
+    Sets-up gridded visibility probability cube
+    """
+    data = np.full((5, 2, 3), dtype=np.float32, fill_value=0.5)
+    cube = set_up_probability_cube(
+        data,
+        thresholds=[50, 1000, 2000, 5000, 7000],
+        variable_name="visibility_in_air",
+        threshold_units="m",
+        spp__relative_to_threshold="less_than",
+        time=datetime(2023, 11, 10, 4, 0),
+    )
+    return cube
+
+
+@pytest.fixture()
+def visibility_spot_cube() -> Cube:
+    """
+    Sets-up spot visibility probability cube
+    """
+    data = np.full((5, 2, 3), dtype=np.float32, fill_value=0.5)
+    cube = set_up_probability_cube(
+        data,
+        thresholds=[50, 1000, 2000, 5000, 7000],
+        variable_name="visibility_in_air",
+        threshold_units="m",
+        spp__relative_to_threshold="less_than",
+        time=datetime(2023, 11, 10, 4, 0),
+    )
+    spot_cube = next(cube.slices_over("latitude"))
+    spot_cube.remove_coord("latitude")
+    spot_cube.coord("longitude").rename("spot_index")
+    return spot_cube
+
+
+@pytest.fixture()
+def cloud_base_gridded_cube() -> Cube:
+    """
+    Sets-up gridded cloud base at ground level probability cube
+    """
+    data = np.full((2, 2, 3), dtype=np.float32, fill_value=0.4)
+    data[0, 0, 0] = 0.7
+    cube = set_up_probability_cube(
+        data,
+        thresholds=[10, 20],
+        variable_name=cloud_name,
+        threshold_units="m",
+        spp__relative_to_threshold="less_than",
+        time=datetime(2023, 11, 10, 4, 0),
+    )
+    cube = cube.extract(
+        iris.Constraint(
+            cloud_base_assuming_only_consider_cloud_area_fraction_greater_than_4p5_oktas=10
+        )
+    )
+    return cube
```

Comment:
You can set up a single threshold cube, but your data shape has to have a leading dimension of size 1, hence the shape below has become `(1, 2, 3)`.
```suggestion
def cloud_base_gridded_cube() -> Cube:
    """
    Sets-up gridded cloud base at ground level probability cube
    """
    data = np.full((1, 2, 3), dtype=np.float32, fill_value=0.4)
    data[0, 0, 0] = 0.7
    cube = set_up_probability_cube(
        data,
        thresholds=[10],
        variable_name=cloud_name,
        threshold_units="m",
        spp__relative_to_threshold="less_than",
        time=datetime(2023, 11, 10, 4, 0),
    )
    return cube
```

Reply:
Ah I couldn't work out how to make it work with one threshold but I wasn't giving it the 1 in the first dimension. I've updated it now.